docker/root/etc/templates/app.ini

```diff
@@ -61,3 +61,6 @@ REQUIRE_SIGNIN_VIEW  = $REQUIRE_SIGNIN_VIEW
 
 [lfs]
 PATH = /data/git/lfs
+
+[git]
+HOME_PATH = /data/git
```

Comment:
Is the change in Docker template files due to my previous confusion regarding container environment? If so, I don't think we should have different values in different environments. That is likely to confuse us and the users in the long run.
Wouldn't this also only apply to fresh installs?

Reply:
When doing upgrade, this will inherit docker user's existing `/data/git`, which is the minimal breaking.

I am fine to either use it or not, just put it here for further discussion.

Reply:
> Is the change in Docker template files due to my previous confusion regarding container environment? 

The docker change is not related to it. The inconsistent should be fixed by "
pass env GNUPGHOME to git comma" https://github.com/go-gitea/gitea/pull/20114/commits/1a386b7f6607f48058b9340eb1212ae8b05b3072  .

Reply:
I was referring to this https://github.com/go-gitea/gitea/pull/20114#discussion_r908020184.
But I got your point. Let's see what the others think about it.

Reply:
Oh ,  sorry , I misunderstood. And yes, your previous comment helps 👍  thank you very much.

Reply:
> If so, I don't think we should have different values in different environments. That is likely to confuse us and the users in the long run.

I've tested the current state and I really don't think it's a good idea to change that in Docker containers. A few days ago I thought it would be great to have it not-that-breaking in Docker environments by setting it to `/data/git`. Let me explain why I changed my mind:

We have two different Docker image variants (root-based and rootless) and there are different ways of using them (simple Docker, docker-compose, Kubernetes, you-name-it).

Based on [root-based setup script](https://github.com/go-gitea/gitea/blob/2921d3c8c9ea17941e4da4381e6fb3177cd7d37d/docker/root/etc/s6/gitea/setup#L17) and [rootless setup script](https://github.com/go-gitea/gitea/blob/2921d3c8c9ea17941e4da4381e6fb3177cd7d37d/docker/rootless/usr/local/bin/docker-setup.sh#L15), the templates are only applied on new installations. For any other Docker users of any kind of deployment this will still be a breaking change as the default path within the code is `%(APP_DATA_PATH)/home`. Our Helm Chart generates the app.ini upfront if non-existing, so even for new Helm Chart users this wouldn't be effective. Imagine new Docker users looking up the default path for that directory: They won't find the "home" path in their environment. There is no hint towards this variable behaving different in Docker images which will most likely confuse us when helping those Docker users identifying issues. And probably the users are confused as well.

@wxiaoguang I'd say, we go the "full breaking change way" and keep the default value for the new setting `%(APP_DATA_PATH)/home` in all environments. Even if this means that users have to adjust their location for the .gitconfig and either relocate their .gnupg folder or set the GNUPGHOME environment variable to keep it there. Having the possibility to change it in app.ini allows for still using the old locations or completely different ones. But that would be up to the users.

Reply:
I see your point. Very thoughtful! Let's revert it.